pkg/monitoring/vmstat/vmstat.go

```diff
@@ -0,0 +1,98 @@
+package vmstat
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type Provider interface {
+	Run() error
+	Shutdown() error
+
+	Name() string
+	IsAvailable() error
+	GetMeasurements() (map[string]interface{}, error)
+}
+
+type provEntry struct {
+	prov Provider
+	wg   sync.WaitGroup
+	run  sync.Once
+}
+
+var providers map[string]*provEntry
```

Comment:
Do we really need global variables?
Usually it's better and cleaner and less error prone if you don't have globals.
Can we crate a type, put these in there and then operate on a type directly?

Reply:
We can do something like, but that again comes to that variable. for that case it's not like a global, more like a `static` analog to C++. Issue is register happens on `init` stage and all that is necessary as some VM implementations are OS specific and we should not allow `*nix` builds to acquire VM provider specific to windows only
```go
type VMStat struct {
}

var vmstat *vmStat
```

Reply:
Ok. I'll pack
Concurrency already being handled